apps/ftu/js/tutorial_utils.js

```diff
@@ -0,0 +1,179 @@
+                    'evt.type: ' + (evt && evt.type));
+        evt.target.removeEventListener('error', onMediaLoadOrError);
+        if (isVideo) {
+          evt.target.removeEventListener('canplay', onMediaLoadOrError);
+          evt.target.removeEventListener('abort', onMediaLoadOrError);
+        } else {
+          evt.target.removeEventListener('load', onMediaLoadOrError);
+        }
+        // Dont block progress on failure to load media
+        if (evt.type === 'error') {
+          console.error('Failed to load tutorial media: ' + src);
+        } else if (evt.type === 'abort') {
+          console.error('Loading of tutorial media aborted: ' + src);
+        }
+        resolve(evt);
+      }
+      function onVideoUnloaded(evt) {
+        console.log('TutorialUtils.loadMedia, onVideoUnloaded for evt.type: ' +
+          (evt && evt.type));
+        mediaElement.removeEventListener('emptied', onVideoUnloaded);
+        mediaElement.addEventListener('canplay', onMediaLoadOrError);
+        mediaElement.addEventListener('abort', onMediaLoadOrError);
+        mediaElement.addEventListener('error', onMediaLoadOrError);
+        console.log('TutorialUtils.loadMedia, onVideoUnloaded, ' +
+                    'assigning src: ' + src);
+        mediaElement.src = src;
+        mediaElement.load();
+      }
+      if (isVideo) {
+        // must unload video and force load before switching to new source
+        if (mediaElement.src) {
+          console.log('TutorialUtils.loadMedia, removing src attribute');
+          mediaElement.removeAttribute('src');
+          mediaElement.addEventListener('emptied', onVideoUnloaded, false);
+          console.log('TutorialUtils.loadMedia, calling load');
+          mediaElement.load();
+        } else {
+          console.log('TutorialUtils.loadMedia, no src, ' +
+            'just calling onVideoUnloaded');
+          onVideoUnloaded();
+        }
+      } else {
+        console.log('TutorialUtils.loadMedia, not a video, ' +
+          'listen for load/error only');
+        mediaElement.addEventListener('load', onMediaLoadOrError, false);
+        mediaElement.addEventListener('error', onMediaLoadOrError, false);
+        mediaElement.src = src;
+      }
+    });
+  },
+
+  /**
+   * Convenience to load and then play video
+   * @param {DOMNode} mediaElement  video or image to assign new src to
+   * @param {String} src  URL for video/image resource
+   * @returns {Promise}
+   */
+  loadAndPlayMedia: function(mediaElement, src) {
+    return this.loadMedia(mediaElement, src).then((evt) => {
+      if (evt.type !== 'error' && mediaElement.nodeName === 'VIDEO') {
+        mediaElement.play();
+      }
+    });
+  },
+
+  /**
+   * Helper to check (via HEAD request) a given resource exists
+   * The returned promise resolves with a true|false value
+   *
+   * @param {String} src  URL for video/image resource
+   * @returns {Promise}
+   */
+  fileExists: function (src) {
```

Comment:
I switched out XHR for fetch here, which does end up much cleaner. But I want callers to be able to rely on a simple true|false result in their callback so I've wrapped fetch's promise here.